src/services/permissionsHelper/Permission.service.js

```diff
@@ -0,0 +1,106 @@
+/* eslint-disable class-methods-use-this */
+const { GeneralError, BadRequest } = require('@feathersjs/errors');
+
+const { copyParams } = require('../../global/utils');
+const { PermissionModel } = require('./models');
+
+const {
+	limitDataViewForReadAccess,
+	restictedAndAddAccess,
+} = require('./hooks');
+
+const permissionServiceHooks = {};
+permissionServiceHooks.before = {
+	all: [restictedAndAddAccess, limitDataViewForReadAccess],
+	find: [],
+	get: [],
+	create: [],
+	update: [],
+	patch: [],
+	remove: [],
+};
+
+
+class PermissionService {
+	constructor(options = {}) {
+		this.docs = options.docs;
+		this.baseServiceName = options.baseService;
+		this.permissionKey = options.permissionKey;
```

Comment:
Sind die sachen immer quierd? dann lieber direkt fordern anstatt von options

Reply:
Da vertrete ich ein anderen Ansatz. In einem Konstruktor einer Klasse sollte immer nur ein Parameter mit gegeben werden. Meistens wird der dann nur opt, c, benannt options habe ich nur aus dem server übernommen. 
Die Klasse wiederum muss sich darum kümmern, das sie für alle Kombinationen die dort gesetzt werden richtig funktioniert. 
new myClass(); new myClass({}); new myClass({ k1, k2, ..});

In Funktionen ist das anders dort können Eingaben erzwungen werden.

Allerdings fehlen hier noch die default Fälle, oder errors beim nicht setzten.

Reply:
hm, naja ist ansichtsache, ich hab schlechte erfahrung mit options gemacht, erstens ist es unspezifisch und erschwert den aufruf, da man selbst nachsehen muss was man setzten muss. Kann man machen wenn man mehere optionen hat die evtl nicht gestzt werden müssen, aber selbst da verleitet es dazu, einer operation zu viel möglich wege zu geben. In dem fall würde ich das ganze so interpretieren, das nicht von den optionen gestzt werdenn muss. Vor allem, da du options auch default mäßig auf {} setzt.